### PR TITLE
feat(CVP-4331): add check-payload tool

### PR DESCRIPTION
**Pull request by yashvardhannanavati**

### CODE REVIEW

**Comment by yashvardhannanavati** [comment]
@dirgim @jsztuka Can I please request your review on this?

**Comment by dirgim** [comment]
@yashvardhannanavati I have reservations about including a go utility into this image which is not geared towards such builds, would having a separate image with which utilizes a supported go builder image instead?
You can consult the [oras-container](https://github.com/konflux-ci/oras-container/tree/main) as an example - this image is also built by Konflux and maintained separately, and is also heavily utilized in build-definitions tasks.

**Comment by dirgim** [comment]
Including also @ralphbean for your opinion on how to include this tool in a best-practice manner.

**Comment by yashvardhannanavati** [comment]
> @yashvardhannanavati I have reservations about including a go utility into this image which is not geared towards such builds, would you consider having a separate image with which utilizes a supported go builder image instead?
> 
> You can consult the [oras-container](https://github.com/konflux-ci/oras-container/tree/main) as an example - this image is also built by Konflux and maintained separately, and is also heavily utilized in build-definitions tasks ([example](https://github.com/konflux-ci/build-definitions/blob/68fe43ef20538da692359f342f6ca1e29ef4825b/task/clair-scan/0.2/clair-scan.yaml#L141)).

Yes, that makes sense. I'll check that out. Thank you!

**Comment by yashvardhannanavati** [comment]
ADR https://konflux-ci.dev/architecture/ADR/0046-common-task-runner-image.html hints at moving towards a common runner image. So, building a separate image for check-payload wouldn't be the right direction as it would introduce overhead in the near future.

After a conversation with @dirgim , we decided to have a separate stage to build the check-payload tool so we can alleviate the concerns pointed out in his previous comment.

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,20 @@
+# Build step for check-payload tool
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.4-1227.1726694542 as check-payload-build
```
[the remark this answers is not in the document]
Comment (yashvardhannanavati) [reply Dockerfile]: that's a good idea. Done!

**Comment by yashvardhannanavati** [comment]
@dirgim @jsztuka @14rcole Could you please help review and merge this? Another build-definitions PR is blocked on this one

**Review by dirgim — APPROVED** [review]
LGTM

**Review by kasemAlem — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,3 +1,14 @@
+# Build step for check-payload tool
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21.13-2.1729776560 as check-payload-build
+
+WORKDIR /opt/app-root/src
+
+ARG CHECK_PAYLOAD_VERSION=0.3.2
+
+RUN curl -k -s -L -o check-payload.tar.gz "https://github.com/openshift/check-payload/archive/refs/tags/${CHECK_PAYLOAD_VERSION}.tar.gz" && \
```
Comment (kdudka) [inline Dockerfile:8]: I do not think that `curl -k` should be used to download data from GitHub.  `-k` means `--insecure`.

Comment (kdudka) [reply Dockerfile:8]: This is being fixed via #349.

**Pull request merged**